src/ray/util/array.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+namespace ray {
+
+// Warning: O(n^2) complexity. Only use in constexpr context.
+template <typename T, size_t N>
+constexpr bool array_is_unique(const std::array<T, N> &arr) {
```

Comment:
Do we want to follow google coding style, which suggests CamelCase for functions? Or we have our own?
Ref: https://google.github.io/styleguide/cppguide.html#Function_Names

Reply:
I tend to do snake cases for free functions for generic util function (that are not specific to Ray) like `debug_string`. It's not a hard rule though

Reply:
I would prefer to have a coding standard somewhere that everyone follows.

> I tend to do snake cases for free functions for generic util function

Seems it's not followed elsewhere, for example https://github.com/ray-project/ray/blob/master/src/ray/gcs/pb_util.h

Reply:
done